block/src/Pos/DB/Block.hs

```diff
@@ -94,18 +94,39 @@ getTipBlock = getTipSomething "block" getBlock
 ----------------------------------------------------------------------------
 
 -- Get serialization of a block with given hash from Block DB.
-getSerializedBlock
-    :: forall ctx m. (MonadRealDB ctx m)
-    => HeaderHash -> m (Maybe ByteString)
-getSerializedBlock hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (blockKeyMod . sformat hashHexF $ hh)
+getSerializedBlock :: MonadRealDB ctx m => HeaderHash -> m (Maybe ByteString)
+getSerializedBlock hh = fmap fst <$> getSerializedBlockOrUndo hh
 
 -- Get serialization of an undo data for block with given hash from Block DB.
-getSerializedUndo :: (MonadRealDB ctx m) => HeaderHash -> m (Maybe ByteString)
-getSerializedUndo hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (undoKeyMod . sformat hashHexF $ hh)
+getSerializedUndo :: MonadRealDB ctx m => HeaderHash -> m (Maybe ByteString)
+getSerializedUndo hh = fmap snd <$> getSerializedBlockOrUndo hh
+
+getSerializedBlockOrUndo ::
+       MonadRealDB ctx m
+    => HeaderHash
+    -> m (Maybe (ByteString, ByteString))
+getSerializedBlockOrUndo hh = do
```

Comment:
Perhaps `getSerializedBlund` or `getSerializedBlockAndUndo` is a more fitting name now?